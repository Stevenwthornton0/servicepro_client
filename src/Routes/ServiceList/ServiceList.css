.serviceList {
    margin-top: 40px;
}

.noneFound {
    font-size: 32px;
    width: fit-content;
    margin-left: auto;
    margin-right: auto;
    margin-top: 150px;
}