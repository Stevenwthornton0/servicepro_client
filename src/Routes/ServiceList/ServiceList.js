import React, { Component } from 'react';
import ServicesApiService from '../../services/services-api-service';
import ServiceListItem from '../../components/ServiceListItem/ServiceListItem';
import AuthApiService from '../../services/auth-api-service';
import TokenService from '../../services/token-service';
import ServicesContext from '../../contexts/ServicesContext';
import LoaderSpinner from '../../Utils/LoaderSpinner';
import FilterList from '../../components/FilterList/FilterList';
import './ServiceList.css';

class ServiceList extends Component {

    static contextType = ServicesContext;

    constructor(props) {
        super(props)
        this.state = {
            serviceType: this.props.match.params.serviceType,
            empty: false
        }
    } 

    componentDidMount() {
        this.context.waitingTrue()
        const { serviceType } = this.state;
        ServicesApiService.getServicesList(serviceType)
            .then(res => {
                    this.context.setServiceList(res)
                    this.context.waitingFalse()
                })
        AuthApiService.getUser(TokenService.getUsername())
        .then(user => {
            this.context.setAdmin(user.admin)
        })
    }

    componentWillUnmount() {
        this.context.clearServiceList()
    }

    deleteService = (serviceId) => {
        const { serviceType } = this.state;
        ServicesApiService.deleteService(serviceId)
            .then(ServicesApiService.getServicesList(serviceType)
                .then(res => {
                        this.context.setServiceList(res)
                    })
                .then(res => 
                    window.location.reload()
                )
            )
    }

    handleServices() {
        const { serviceList, isAdmin } = this.context;
        if (serviceList.length) {
            return (
                    serviceList.map(service => 
                        <ServiceListItem 
                            key={service.id}
                            service={service}
                            admin={isAdmin}
                            deleteService={this.deleteService}
                        />
                    )
            )
        } if (!serviceList.length) {
            return (
                <h3 className='noneFound'>No Services Found</h3>
            )
        }
    }

    refreshPage = () => {
        window.location.reload()
    }

    filterPage = ev => {
        ev.preventDefault()
        this.context.waitingTrue();
        let { city, state } = ev.target;
        city = city.value.toLowerCase();
        state = state.value.toLowerCase();

        const { serviceType } = this.state;
        ServicesApiService.getServicesList(serviceType, city, state)
            .then(res => {
                console.log(res)
                    this.context.setServiceList(res)
                    this.context.waitingFalse()
                })
    }

    render () {
        const { waiting } = this.context;
        return (
            <div className='serviceList'>
                <div className='loader'>
                    {waiting && <LoaderSpinner />}
                </div>

                {!waiting && <FilterList handleFilter={this.filterPage} handleClear={this.refreshPage}/>}
                {!waiting && this.handleServices()}
            </div>
        )
    }
}

export default ServiceList;