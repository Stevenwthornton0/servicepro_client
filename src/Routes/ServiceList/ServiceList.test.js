import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme'
import ServiceList from './ServiceList';

Enzyme.configure({ adapter: new Adapter() })

describe('ServiceList component', () => {
    it('renders without crashing', () => {
        const match = {
            params: {
                serviceType: "mechanic"
            }
        }
        const wrapper = mount(<ServiceList match={match}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})