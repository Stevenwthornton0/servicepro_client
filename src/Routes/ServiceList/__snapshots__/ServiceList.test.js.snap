// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceList component renders without crashing 1`] = `
<ServiceList
  match={
    Object {
      "params": Object {
        "serviceType": "mechanic",
      },
    }
  }
>
  <div
    className="serviceList"
  >
    <div
      className="loader"
    />
    <FilterList
      handleClear={[Function]}
      handleFilter={[Function]}
    >
      <div
        className="filter"
      >
        <p
          className="bold"
        >
          Filter by:
        </p>
        <form
          className="filterForm"
          onSubmit={[Function]}
        >
          <div>
            <label>
              City
            </label>
            <input
              id="filterList_city"
              name="city"
              required={true}
            />
          </div>
          <div>
            <label>
              State
            </label>
            <States>
              <select
                className="state"
                id="state"
                name="state"
                required={true}
              >
                <option
                  value=""
                >
                  ---
                </option>
                <option
                  value="Alabama"
                >
                  Alabama
                </option>
                <option
                  value="Alaska"
                >
                  Alaska
                </option>
                <option
                  value="Arizona"
                >
                  Arizona
                </option>
                <option
                  value="Arkansas"
                >
                  Arkansas
                </option>
                <option
                  value="California"
                >
                  California
                </option>
                <option
                  value="Colorado"
                >
                  Colorado
                </option>
                <option
                  value="Connecticut"
                >
                  Connecticut
                </option>
                <option
                  value="Delaware"
                >
                  Delaware
                </option>
                <option
                  value="District of Columbia"
                >
                  District of Columbia
                </option>
                <option
                  value="Florida"
                >
                  Florida
                </option>
                <option
                  value="Georgia"
                >
                  Georgia
                </option>
                <option
                  value="Hawaii"
                >
                  Hawaii
                </option>
                <option
                  value="Idaho"
                >
                  Idaho
                </option>
                <option
                  value="Illinois"
                >
                  Illinois
                </option>
                <option
                  value="Indiana"
                >
                  Indiana
                </option>
                <option
                  value="Iowa"
                >
                  Iowa
                </option>
                <option
                  value="Kansas"
                >
                  Kansas
                </option>
                <option
                  value="Kentucky"
                >
                  Kentucky
                </option>
                <option
                  value="Louisiana"
                >
                  Louisiana
                </option>
                <option
                  value="Maine"
                >
                  Maine
                </option>
                <option
                  value="Maryland"
                >
                  Maryland
                </option>
                <option
                  value="Massachusetts"
                >
                  Massachusetts
                </option>
                <option
                  value="Michigan"
                >
                  Michigan
                </option>
                <option
                  value="Minnesota"
                >
                  Minnesota
                </option>
                <option
                  value="Mississippi"
                >
                  Mississippi
                </option>
                <option
                  value="Missouri"
                >
                  Missouri
                </option>
                <option
                  value="Montana"
                >
                  Montana
                </option>
                <option
                  value="Nebraska"
                >
                  Nebraska
                </option>
                <option
                  value="Nevada"
                >
                  Nevada
                </option>
                <option
                  value="New Hampshire"
                >
                  New Hampshire
                </option>
                <option
                  value="New Jersey"
                >
                  New Jersey
                </option>
                <option
                  value="New Mexico"
                >
                  New Mexico
                </option>
                <option
                  value="New York"
                >
                  New York
                </option>
                <option
                  value="North Carolina"
                >
                  North Carolina
                </option>
                <option
                  value="North Dakota"
                >
                  North Dakota
                </option>
                <option
                  value="Ohio"
                >
                  Ohio
                </option>
                <option
                  value="Oklahoma"
                >
                  Oklahoma
                </option>
                <option
                  value="Oregon"
                >
                  Oregon
                </option>
                <option
                  value="Pennsylvania"
                >
                  Pennsylvania
                </option>
                <option
                  value="Rhode Island"
                >
                  Rhode Island
                </option>
                <option
                  value="South Carolina"
                >
                  South Carolina
                </option>
                <option
                  value="South Dakota"
                >
                  South Dakota
                </option>
                <option
                  value="Tennessee"
                >
                  Tennessee
                </option>
                <option
                  value="Texas"
                >
                  Texas
                </option>
                <option
                  value="Utah"
                >
                  Utah
                </option>
                <option
                  value="Vermont"
                >
                  Vermont
                </option>
                <option
                  value="Virginia"
                >
                  Virginia
                </option>
                <option
                  value="Washington"
                >
                  Washington
                </option>
                <option
                  value="West Virginia"
                >
                  West Virginia
                </option>
                <option
                  value="Wisconsin"
                >
                  Wisconsin
                </option>
                <option
                  value="Wyoming"
                >
                  Wyoming
                </option>
              </select>
            </States>
          </div>
          <div
            id="butContainer"
          >
            <button
              type="submit"
            >
              Filter
            </button>
            <button
              onClick={[Function]}
            >
              Clear
            </button>
          </div>
        </form>
      </div>
    </FilterList>
    <h3
      className="noneFound"
    >
      No Services Found
    </h3>
  </div>
</ServiceList>
`;
